src/ucp/wireup/wireup.c

```diff
@@ -1297,11 +1297,14 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep, ucp_ep_config_key_t *new_key,
             ucp_ep_set_lane(ep, lane, NULL);
         }
 
-        ucs_assert(ucp_ep_get_lane(ep, lane) == NULL);
+        ucs_assert(ucp_ep_get_lane(ep, lane, 0) == NULL);
     }
 
-    ucs_assert(sizeof(new_uct_eps) == sizeof(ep->uct_eps));
-    memcpy(ep->uct_eps, new_uct_eps, sizeof(ep->uct_eps));
+    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
```

Comment:
to be same as the memcpy: 
1. iterate until UCP_MAX_LANES
2. don't check for NULL